linked_list.py

```diff
@@ -0,0 +1,60 @@
+
+#import numpy as np
+
+# insertAt
+# deleteAt
+# first
+# last
+
+class Node:
+    def __init__(self, data=None):
+        self.next = None
+        self.data = data
+
+class LinkedList:
+    def __init__(self):
+        self.tail = None
+        self.head = None
+        self.size = 0
+
+    def append(self, data):
+        if self.size == 0:
+            self.head = Node(data)
+            self.tail = self.head
+        else:
+            node = Node(data)
+            self.tail.next = node
+            self.tail = self.tail.next
+        self.size += 1
+
+    def delete(self):
+        node = self.head
+        while node is not None:
+            if node.next==self.tail:
+                node.next = None
+                self.tail = node
```

Comment:
You can make this clear such as 

self.tail = None (Instead of referencing it to node which is guaranteed to be None in the previous line)

Reply:
ignore